Doesn't compile with Qt 5.8 RC

<!--- Provide a general summary of the issue in the title above -->
On building Keepassxc on Arch Linux from the AUR package keepassxc-git with the  5.8 Release Candidate version of QT (from Arch's `testing` repository), the following message pops up while compiling:

```[ 29%] Building CXX object src/CMakeFiles/keepassx_core.dir/format/KeePass2XmlReader.cpp.o                                                                              
/home/dominik/.tmp/yaourt/yaourt-tmp-dominik/aur-keepassxc-git/src/keepassxc/src/format/KeePass2XmlReader.cpp: In member function 'void KeePass2XmlReader::parseBinaries()':
/home/dominik/.tmp/yaourt/yaourt-tmp-dominik/aur-keepassxc-git/src/keepassxc/src/format/KeePass2XmlReader.cpp:391:77: error: call of overloaded 'compare(const char [5], Qt::CaseSensitivity)' is ambiguous
           if (attr.value("Compressed").compare("True", Qt::CaseInsensitive) == 0) {
                                                                           ^
In file included from /usr/include/qt/QtCore/qregexp.h:47:0,
                 from /usr/include/qt/QtCore/qstringlist.h:47,
                 from /usr/include/qt/QtCore/qlist.h:1093,
                 from /usr/include/qt/QtCore/qstringlist.h:41,
                 from /usr/include/qt/QtGui/qcolor.h:46,
                 from /usr/include/qt/QtGui/QColor:1,
                 from /home/dominik/.tmp/yaourt/yaourt-tmp-dominik/aur-keepassxc-git/src/keepassxc/src/format/KeePass2XmlReader.h:21,
                 from /home/dominik/.tmp/yaourt/yaourt-tmp-dominik/aur-keepassxc-git/src/keepassxc/src/format/KeePass2XmlReader.cpp:18:
/usr/include/qt/QtCore/qstring.h:1584:12: note: candidate: int QStringRef::compare(const QString&, Qt::CaseSensitivity) const
 inline int QStringRef::compare(const QString &s, Qt::CaseSensitivity cs) const Q_DECL_NOTHROW
            ^~~~~~~~~~
/usr/include/qt/QtCore/qstring.h:1515:9: note: candidate: int QStringRef::compare(const QByteArray&, Qt::CaseSensitivity) const
     int compare(const QByteArray &s, Qt::CaseSensitivity cs = Qt::CaseSensitive) const
         ^~~~~~~
make[2]: *** [src/CMakeFiles/keepassx_core.dir/build.make:903: src/CMakeFiles/keepassx_core.dir/format/KeePass2XmlReader.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:257: src/CMakeFiles/keepassx_core.dir/all] Error 2
make: *** [Makefile:139: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
==> ERROR: Makepkg was unable to build keepassxc-git.
```

Relevant output from `qtdiag-qt5` is as follows:

```
Qt 5.8.0 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 6.3.1 20170109) on "xcb"                                                                  
OS: Arch Linux [linux version 4.9.5-1-ARCH]                                                                                                                          
                                                                                                                                                                     
Architecture: x86_64; features: SSE2 SSE3 SSSE3                                                                                                                      
                                                                                                                                                                     
Environment:                                                                                                                                                         
  QT_AUTO_SCREEN_SCALE_FACTOR="0"                                                                                                                                    
                                                                                                                                                                     
Features: QT_NO_EXCEPTIONS  
```

I'm not sure whether this issue will continue occurring in Qt 5.8 proper, but I've seen this noted on [the keepassxc AUR page](https://aur.archlinux.org/packages/keepassxc/) and felt it was best to note it here.

--- a/src/format/KeePass2XmlReader.cpp
+++ b/src/format/KeePass2XmlReader.cpp
@@ -388,7 +388,7 @@ void KeePass2XmlReader::parseBinaries()
             QString id = attr.value("ID").toString();
 
             QByteArray data;
-            if (attr.value("Compressed").compare("True", Qt::CaseInsensitive) == 0) {
+            if (attr.value("Compressed").compare(QLatin1String("True"), Qt::CaseInsensitive) == 0) {
                 data = readCompressedBinary();
             }
             else {

